### DISCUSSION TITLE
Ports to other languages

### DISCUSSION THREAD

**Discussion by hunar4321** [hunar4321/particle-life#48] [Announcements]
Hi Everyone

If you have ported the code to other languages and you like your project to be linked with this one. I'll be more than happy to put the link of your project in the readme section.

Best

**Comment by CapsAdmin** [comment]
I made a high performant 2d particle simulation in WebGL computing on the GPU!

https://capsadmin.github.io/webgl-particles/

Click the hamburger icon in the top left corner and select particle life from the dropdown menu.

It's not 100% the same, but I included the ability to use javascript code in the CONFIG block that generates the rules similar to how you do it.

I get 60 fps on an iphone 12 with 7 colors and 500 particles for each color.

Source code can be found here:
https://github.com/CapsAdmin/webgl-particles

**Reply by hunar4321** [reply]
Great Job. I'll add the GitHub link to Other ports section.
Keep up the good work

**Comment by BlinkSun** [comment]
Hi @hunar4321 ,

I did a port of the simple javascript version into C# Winforms (GDI+) (.Net Core 6)

https://github.com/BlinkSun/ParticleLifeSimulation

I ll continue to push some update to my repo to add more feature in next days/weeks like yours.

Thanks for your inspiration and continue your great work !

BlinkSun

**Reply by hunar4321** [reply]
Well done. I have added it to the Other ports.

**Comment by helloimalemur** [comment]
Hello Professor,

I was very inspired by your project and youtube video. So I decided to create my own port of the java script version, and to my delight it doesn't look like anyone else had written it in Java yet! Here's my port written in Java, using the Game Loop design model, and built using gradle. I do intend on adding gui features such as those present in your c++ version. I would love to have it included in your list. 
https://github.com/helloimalemur/ParticleSimulation

Thank you,
James

**Reply by hunar4321** [reply]
Well done 👍 
Added

**Reply by helloimalemur** [reply]
Thank you!

**Comment by youssefboulmalf** [comment]
Hey, 

I made a golang version of your project with Pixels. Only after creation I found out someone already made a golang port :( . 

This is my first golang project and I am currently working on a particle fuzing feature after that I plan on adding controls to the program.

Thanyou for creating this project and inspiring me!

[Particle-Life-Go](https://github.com/youssefboulmalf/Particle-Life-Go)

Youssef

**Reply by hunar4321** [reply]
Well done. Keep it up

**Comment by richgain** [comment]
Hi there,
I have completed a port of the javascript code designed to run in Arduino code on the ESP32 T-Display S3.
I hope someone else might find it interesting.

https://github.com/richgain/Particle-life-arduino/tree/main

https://youtu.be/OT6iyhJBm8Q

richgain

**Comment by danieldemac** [comment]
Hello Professor! Your project inspired me to create something similar. I used the base code and made a simulation in HTML as well as in UNITY3D. I will organize a repository and make a video in Brazilian Portuguese. I will share it here when it's ready! (ps. I presented this work at a technology fair at my college, and my professors loved it!)

**Reply by hunar4321** [reply]
Great. Keep it up
